js/dydx.js

```diff
@@ -0,0 +1,1252 @@
+                'fetchOrder': true,
+                'fetchOpenOrders': true,
+                'fetchCanceledOrders': true,
+                'fetchClosedOrders': false,
+                'fetchOrderBook': true,
+                'fetchOrders': true,
+                'fetchOrderTrades': true,
+                'fetchPosition': true,
+                'fetchPositions': true,
+                'fetchTicker': true,
+                'fetchTickers': true,
+                'fetchTime': true,
+                'fetchTrades': true,
+                'fetchTradingFees': undefined,
+                'fetchTransactions': true,
+                'fetchWithdrawals': false,
+                'privateAPI': true,
+                'publicAPI': true,
+                'withdraw': undefined, // TODO, needs stark sign
+            },
+            'timeframes': {
+                '1m': '1MIN',
+                '5m': '5MINS',
+                '15m': '15MINS',
+                '30m': '30MINS',
+                '1h': '1HOUR',
+                '4h': '4HOURS',
+                '1d': '1DAY',
+            },
+            'urls': {
+                'logo': '',
+                'test': {
+                    'public': 'https://api.stage.dydx.exchange',
+                    'private': 'https://api.stage.dydx.exchange',
+                },
+                'api': {
+                    'public': 'https://api.dydx.exchange',
+                    'private': 'https://api.dydx.exchange',
+                },
+                'www': 'https://trade.dydx.exchange/',
+                'www-test': 'https://trade.stage.dydx.exchange/', // TODO: which key should be for test-websites?
+                'doc': [
+                    'https://docs.dydx.exchange/',
+                ],
+                'referral': '',
+            },
+            'api': {
+                'public': {
+                    'get': {
+                        'markets': 60,
+                        'orderbook/{market}': 60,
+                        'trades/{market}': 60,
+                        'fast-withdrawals': 60, // implicit
+                        'stats/{market}': 60,
+                        'historical-funding/{market}': 60,
+                        'candles/{market}': 60,
+                        'config': 60, // Get default maker and taker fees.
+                        'users/exists': 60,
+                        'usernames': 60,
+                        'time': 60,
+                        'leaderboard-pnl': 60,
+                    },
+                    'put': {
+                        'emails/verify-email': 60,
+                    },
+                },
+                'private': {
+                    'get': {
+                        'rewards/public-retroactive-mining': 60, // TODO: write to exchange API team, that this endpoint is listed under 'public' endpoints, but needs apikey
+                        'rewards/retroactive-mining': 60,
+                        'active-orders': 1, // variable
```

Comment:
What does `variable` mean in this case ?

If it depends on some arguments, we might have to implement `calculateRateLimiterCost`: 

- https://github.com/ccxt/ccxt/blob/master/js/bitvavo.js#L109
- https://github.com/ccxt/ccxt/blob/master/js/bitvavo.js#L1535

- https://github.com/ccxt/ccxt/blob/master/js/binance.js#L445
- https://github.com/ccxt/ccxt/blob/master/js/binance.js#L5229

Reply:
I'll address that tomorrow. I am not well-versed in rate-limiter inners and I'll have to familiarize with its internals at first.